doc/contribute/contributor_expectations.rst

```diff
@@ -87,27 +87,37 @@ Changes which require an RFC proposal include:
 Maintainers have the discretion to request that contributors create an RFC for
 PRs that are too large or complicated.
 
+.. _pr_requirements:
+
 PR Requirements
 ***************
 
 - Each commit in the PR must provide a commit message following the
   :ref:`commit-guidelines`.
 
+- No fixup or merge commits are allowed, see :ref:`Contribution workflow` for
+  more information.
+
 - The PR description must include a summary of the changes and their rationales.
 
 - All files in the PR must comply with :ref:`Licensing
   Requirements<licensing_requirements>`.
 
-- Follow the Zephyr :ref:`coding_style` and :ref:`coding_guidelines`.
+- The code must follow the Zephyr :ref:`coding_style` and :ref:`coding_guidelines`.
 
-- PRs must pass all CI checks. This is a requirement to merge the PR.
+- The PR must pass all CI checks, as described in :ref:`merge_criteria`.
   Contributors may mark a PR as draft and explicitly request reviewers to
   provide early feedback, even with failing CI checks.
 
 - When breaking a PR into multiple commits, each commit must build cleanly. The
   CI system does not enforce this policy, so it is the PR author's
   responsibility to verify.
 
+- When breaking up a PR into multiple commits, the sequence of commits must
+  follow a logical order as per the changes introduced, and not reflect the
+  order in which the changes were developed or modified. This includes not
+  renaming or rewriting files that were introduced in the PR itself.
```

Comment:
If you want to contribute something it is up to you to get it into shape for acceptance to the project, not up to others

Reply:
I disagree in this particular case. Adding a file and then renaming it has been a recipe for Request for Changes forever in this project, at least in my personal experience. Hence in this case we are only writing down an unwritten rule, which justifies it in my opinion. I doubt this will have the effect that @ajf58 mentions, because I have not seen it happen these past years at all.